hpx/parallel/executors/persistent_auto_chunk_size.hpp

```diff
@@ -0,0 +1,126 @@
+        ///
+        /// \note Default constructed \a persistent_auto_chunk_size executor parameter
+        ///       types will use 80 microseconds as the minimal time for which
+        ///       any of the scheduled chunks should run.
+        ///
+
+
+        /// Construct an \a persistent_auto_chunk_size executor parameters object
+        ///
+        /// \param rel_time     [in] The time duration to use as the minimum
+        ///                     to decide how many loop iterations should be
+        ///                     combined.
+        ///
+
+        explicit persistent_auto_chunk_size(hpx::util::steady_duration time_cs = boost::chrono::nanoseconds(0))
+          : chunk_size_time_(time_cs.value().count())
+        {}
+
+        /// \cond NOINTERNAL
+        // Estimate a chunk size based on number of cores used.
+        template <typename Executor, typename F>
+        std::size_t get_chunk_size(Executor& exec, F && f, std::size_t count)
+        {
+            std::size_t const cores = executor_information_traits<Executor>::
+                processing_units_count(exec, *this);
+
+            if (count > 100*cores)
+            {
+                boost::uint64_t t;
+                if(chunk_size_time_ == 0)
+                {
+                    using hpx::util::high_resolution_clock;
+                    t = high_resolution_clock::now();
+
+                    std::size_t test_chunk_size = f();
+                    if (test_chunk_size != 0)
+                    {
+                        
+                        t = (high_resolution_clock::now() - t) / test_chunk_size;
+                        chunk_size_time_ = t;
+                    
+                        if (t != 0)
+                        {
+                            // return chunk size which will create the required
+                            // amount of work
+                            return (std::min)(count, (std::size_t)(min_time_ / t));
+                        }
+                    }
+                }
+
+                else
+                {
+                    t = chunk_size_time_;
+                    return (std::min)(count, (std::size_t)(min_time_ / t));
+                }
+            }
+
+            
+            return (count + cores - 1) / cores;
+            
+        }
+        /// \endcond
+
+        std::size_t get_chunk_size_value() const
+        {
+        return chunk_size_time_;
+        }
+
+    private:
+        /// \cond NOINTERNAL
+        friend class hpx::serialization::access;
+
+        template <typename Archive>
+        void serialize(Archive & ar, const unsigned int version)
+        {
+            ar  & chunk_size_time_& min_time_;
+        }
+        /// \endcond
+
+    private:
+        /// \cond NOINTERNAL
+        boost::uint64_t chunk_size_time_;
+        static constexpr boost::uint64_t min_time_ = 80000;      // nanoseconds
+        /// \endcond
```

Comment:
Please use `HPX_CONSTEXPR` instead of `constexpr` (some compilers might not do the right things otherwise).